### PR TITLE
Do not enable UART DMA flags unconditionally

### LINKED ISSUE #334: UARTs should not enable DMA by default
In https://github.com/stm32-rs/stm32f1xx-hal/blob/5d55f83528a1a4eebcc672201751a320d9be5854/src/serial.rs#L312, every U(S)ART enables the DMA transmit and receive bits (dmat and dmar) unconditionally when initializing the UART struct.

This interferes with other users of the DMA channels, and it appears to me that this should rather be done in `TransferPayload::start()/::stop()`: https://github.com/stm32-rs/stm32f1xx-hal/blob/5d55f83528a1a4eebcc672201751a320d9be5854/src/serial.rs#L530-L546), like in the SPI implementation https://github.com/stm32-rs/stm32f1xx-hal/blob/5d55f83528a1a4eebcc672201751a320d9be5854/src/spi.rs#L569-L586

Or did I overlook something here?

### PR DESCRIPTION
**Pull request by Windfisch**
Fixes #334

I have not tested that code, though, so proceed with caution :)

### CODE REVIEW

**Comment by Windfisch** [comment]
Updated as per https://github.com/stm32-rs/stm32f1xx-hal/issues/334#issuecomment-842916316.

**Comment by Windfisch** [comment]
Renamed serial-dma `split` and i2c `free` functions to `release` as per https://github.com/stm32-rs/stm32f1xx-hal/issues/334#issuecomment-843724427.

File: src/serial.rs
Comment (TheZoq2) [inline src/serial.rs:567]: Could you remove the `mut` here since it gives a warning now that self is just passed along?

File: src/serial.rs
Comment (TheZoq2) [inline src/serial.rs:583]: Remove mut here as well

**Review by TheZoq2 — CHANGES_REQUESTED** [review]
Thanks for the PR! I pointed out a few `mut`s which cause warnings, other than that I think this is ready to merge (possibly after testing on hw)

**Comment by burrbull** [comment]
cc @TeXitoi

**Review by TeXitoi — APPROVED** [review]

**Comment by burrbull** [comment]
> @burrbull you want acces right to this repo?

It would be great.

**Comment by TeXitoi** [comment]
I can't find how to add you to the stm32f1 team, looks like I don't have the right or I can't find the settings

**Comment by adamgreig** [comment]
Previously only @therealprof was a maintainer for the stm32f1 team, but I've added you as well @TeXitoi, so you should be able to add new team members now.

**Comment by TeXitoi** [comment]
Thanks, @burrbull added.

**Comment by burrbull** [comment]
Thanks.

**Pull request merged**
